scripts/release/asset_publish.py

```diff
@@ -169,14 +169,24 @@ def validate_and_prepare_pipeline_component(
     :rtype: bool
     """
     with open(spec_path) as f:
-        pipeline_dict = yaml.safe_load(f)
+        try:
+            pipeline_dict = yaml.safe_load(f)
+        except Exception:
+            logger.log_error(f"Error in loading component spec at {spec_path}")
+            return False
 
     jobs = pipeline_dict['jobs']
     logger.print(f"Preparing pipeline component {pipeline_dict['name']}")
     updated_jobs = {}
 
     for job_name, job_details in jobs.items():
         logger.print(f"job {job_name}")
+        if not job_details.get('component'):
+            # if-else or inline component
+            logger.println(f"component not defined for job {job_name}")
```

Comment:
Should this be a warning, though?
```suggestion
            logger.print(f"component not defined for job {job_name}")
```